cmd/monaco/deploy/deploy.go

```diff
@@ -124,26 +126,80 @@ func loadProjects(fs afero.Fs, manifestPath string, man *manifest.Manifest, spec
 }
 
 func checkEnvironments(projects []project.Project, envs manifest.Environments) error {
+	errs := []error{}
+	openPipelineKindCoordinatesPerEnvironment := map[string]map[string][]coordinate.Coordinate{}
 	for _, p := range projects {
 		for envName, cfgPerType := range p.Configs {
-			if _, found := envs[envName]; !found {
-				return fmt.Errorf("cannot find environment `%s`", envName)
+			env, found := envs[envName]
+			if !found {
+				errs = append(errs, fmt.Errorf("undefined environment %q", envName))
+				continue
 			}
-			for _, cfgs := range cfgPerType {
-				if err := checkConfigsForEnvironment(envs[envName], cfgs); err != nil {
-					return err
+
+			openPipelineKindCoordinates := openPipelineKindCoordinatesPerEnvironment[envName]
+			if openPipelineKindCoordinates == nil {
+				openPipelineKindCoordinates = make(map[string][]coordinate.Coordinate)
+			}
+			cfgPerType.ForEveryConfigDo(func(cfg config.Config) {
+				if cfg.Skip {
+					return
+				}
+
+				if openPipelineType, ok := cfg.Type.(config.OpenPipelineType); ok {
+					coordinates := openPipelineKindCoordinates[openPipelineType.Kind]
+					coordinates = append(coordinates, cfg.Coordinate)
+					openPipelineKindCoordinates[openPipelineType.Kind] = coordinates
 				}
+			})
+			openPipelineKindCoordinatesPerEnvironment[envName] = openPipelineKindCoordinates
+
+			if err := checkIfConfigsForEnvironmentRequireUnavailablePlatform(env, cfgPerType); err != nil {
+				errs = append(errs, err)
 			}
 		}
 	}
-	return nil
+
+	for envName, openPipelineKindCoordinates := range openPipelineKindCoordinatesPerEnvironment {
+		for kind, coordinates := range openPipelineKindCoordinates {
+			if len(coordinates) > 1 {
+				errs = append(errs, fmt.Errorf("environment %q has multiple OpenPipeline configurations of kind %q: %s", envName, kind, coordinateSliceAsString(coordinates)))
+			}
+		}
+	}
```

Comment:
This could also be extracted into a function call e.g., `checkOpenPipelineConfigs(openPipelineCoordinates) []error`